students/MicahBraun/Lesson 05/mailroom_3.py

```diff
@@ -0,0 +1,297 @@
+    first_names = []  # first names separated out to be used for letters
+    default = os.getcwd()  # get and assign current directory to variable
+    default = default + "\\"  # format default so that it will be accepted (will work with Windows)
+    while True:
+        msg = input("Proceed to write letters to all donors in donor list? (Y/N) ")  # if user selects Y, proceed
+        if msg.upper() == "Y":  # check input
+            try:
+                working_directory = str(input('\nEnter the file path where you want to write the letters - '
+                                              'don\'t forget to use ''two \n\'\\\\\' as file separators to make sure'
+                                              ' it is compatible if using Windows - (e.g. \'C:\\\\\': '))
+            except Exception as e:              # exception handler - prints line of error, and type
+                print('Error on line {}'.format(sys.exc_info()[-1].tb_lineno), type(e).__name__, e)
+            # working directory variable assignment by user - if blank, use default
+            if working_directory == "":  # check for input on working_directory
+                working_directory = default
+
+            for x in range(int(len(donor_dict))):  # for items in the span of donor_dict
+                firstname = list(donor_dict.values())[x][0]  # firstname = donor_dict at index[item], column[0]
+                firstname = firstname.split(' ', ).pop(0)  # format var to split on whitespace (only first name)
+                first_names.append(firstname)  # append this value to empty list first_names
+
+            for i in range(int(len(list(donor_dict.values())))):  # for items in the span of donor_dict
+                elem = list(donor_dict.values())[i][0]  # elem = donor_dict at index[i], column[0]
+                elem = elem.replace(" ", "")  # format name w/o whitespace
+                names_list.append(elem)  # append to name_list
+
+            for names in range(1, int(len(names_list))):  # omit header (range 0), for each name in names_list[]
+                try:
+                    with open(working_directory + str(names_list[names]) + '.txt',
+                              'a') as filename:  # create files/file-names
+                        filename.write(today)  # write current date (mm/dd/yy)
+                        filename.write('\n\nDear {0}, \n\nThank you for your continued support through your contribution '
+                                       'of {1} towards our Foundation\'s fundraising goal.\n\nBest wishes,\n'
+                                       'Foundation Board of Directors'
+                                       '\n'.format(str(first_names[names]),
+                                                   locale.currency(list(donor_dict.values())[names][1], grouping=True)))
+                except Exception as e:              # exception handler - prints line of error, and type
```

Comment:
Same issue here, you want to try and catch a specific anticipate error. Try producing a specific exception in your code and then catch that here